src/user.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * @fileoverview Genotet user.
+ */
+
+'use strict';
+
+/** @const */
+genotet.user = {};
+
+/**
+ * @typedef {{
+ *   username: (string|undefined),
+ *   password: string,
+ *   sessionId: (string|undefined),
+ *   expiration: (string|undefined)
+ * }}
+ */
+genotet.Cookie;
+
+/**
+ * Initializes the user auth.
+ */
+genotet.user.init = function() {
+  if (!document.cookie ||
+    new Date().getTime() >= Cookies.get('expiration')) {
+    return;
+  }
+
+  var userInfo = {
+    type: 'sign-in',
+    username: Cookies.get('username'),
+    password: Cookies.get('password')
+  };
+
+  $.ajax({
+    url: genotet.data.userURL,
+    type: 'POST',
+    data: userInfo,
+    dataType: 'json'
+  }).done(function(data) {
+    genotet.menu.displaySignedUser(userInfo.username);
+    genotet.data.userInfo = {
```

Comment:
pick returns a copy of the object, filtered to only have values for the whitelisted keys, and also copied the type of original object to the new object. However, data.cookie and genotet.data.userInfo are defined as different types. So I think it is not a good idea to use pick here.